app/frontend/wallet/explorer-types.ts

```diff
@@ -0,0 +1,169 @@
+export type HostedPoolMetadata = {
+  name: string
+  description: string
+  ticker: string
+  homepage: string
+  extended?: string
+}
+
+export type StakePoolInfoExtended = {
+  pledge: string
+  margin: number
+  fixedCost: string
+  name: string
+  ticker: string
+  homepage: string
+  poolHash: string
+  liveStake: string
+  roa: string
+  epochBlocks: string
+  lifeTimeBlocks: string
+  saturatedPercentage: number
+}
+
+export type DelegationHistoryEntry = StakePoolInfoExtended & {
+  txHash: string
+  time: string
+  epochNo: number
+}
+
+export type RewardsHistoryEntry = StakePoolInfoExtended & {
+  time: string
+  epochNo: number
+  forDelegationInEpoch: number
+  amount: string
+}
+
+export type WithdrawalsHistoryEntry = {
+  time: string
+  epochNo: number
+  txHash: string
+  amount: string
+}
+
+export type StakeRegistrationHistoryEntry = {
+  txHash: string
+  time: string
+  epochNo: number
+  action: 'registration' | 'deregistration'
+}
+
+export type ValidStakePools = {
```

Comment:
I'd name this `ValidStakePoolsMapping` to make it clear it's not an array/collection and I would create a separate type for the object value, something like `StakePoolProps`

The ideal endgame would imho be just returning a list of stake pools and whoever needs to look them up by id could just sequentially search it or delegate it to some function which would be optimized for that should that be an actual bottleneck

Reply:
Done. Originally, we went for the Map due to many places which needed to access it, and at that time poolId was sufficient. Therefore we decided that O(1) lookup by just the poolId would be ideal for the time being.

Reply:
btw @DavidTranDucVL is already working on a refactor regarding this validStakePools Object btw, so we address any suggestions in that PR then.